app/src/lib/branches/dragActions.ts

```diff
@@ -0,0 +1,53 @@
+import { DraggableCommit, DraggableHunk, DraggableFile } from '$lib/dragging/draggables';
+import { filesToOwnership } from '$lib/vbranches/ownership';
+import type { BranchController } from '$lib/vbranches/branchController';
+import type { Branch } from '$lib/vbranches/types';
+
+class BranchDragActions {
+	constructor(
+		private branchController: BranchController,
+		private branch: Branch
+	) {}
```

Comment:
I was thinking about whether I should have some
```ts
interface DropzoneActions {
  accepts: //...
  onDrop: //...
}
```
So we'd have one class for each dropzone actions. It would also mean that we could pass something that implements that interface directly to the Dropzone component, but I didn't quite want to go that far yet.

Reply:
Yeah, makes sense. Definitely sounds like a good next step. However as you said, lets maybe save that for a more full-scale refactor of the whole drag-n-drop system :+1: